### ISSUE TITLE
Suggestion for Catboost Tutorial

### ISSUE THREAD

**Issue by dennislamcv1** [catboost/catboost#1117]
Problem: Request for a Catboost Tutorial for Regression problems
catboost version: Any version
Operating System: WIndows
CPU: i7
# GPU: None

Hi Yandex, I am currently learning how to use Catboost for ML projects. Would love to have a tutorial on Regression problems using real data set consists of mixture of categorical and numerical features.

Please do not use those generic datasets like Boston Housing et al. You can select one proper dataset from Kaggle et al as an example. 

The tutorial must geared for first time users hence comments and guidance in Jupyter Notebook will be helpful.

Thanks much.

**Comment by annaveronika** [comment]
This is a great idea for contributions!

**Comment by dvddn** [comment]
Is this still relevant?
I'd love to contribute.

**Comment by annaveronika** [comment]
Yes, tutorials are always welcome! We have tutorials classification problems, but I think we don't have one for regression. So please, contribute!

**Comment by dennislamcv1** [comment]
There is one data scientist did a show: [https://www.viralml.com/video-content.html?v=zKEPXQzqxe0&Title=CatBoost%20vs%20XGBoost%20-%20A%20Gentle%20Introduction%20to%20CatBoost%20-%20Free%20Udemy%20Class](url)

**Comment by dennislamcv1** [comment]
[CatBoost vs XGBoost Battle & CatBoost Introduction.pdf](https://github.com/catboost/catboost/files/4233199/CatBoost.vs.XGBoost.Battle.CatBoost.Introduction.pdf)

**Comment by dennislamcv1** [comment]
How to upload Jupyter notebooks in this thread?

**Comment by annaveronika** [comment]
You could make a tutorial with Boston out of this battle for regression.
I looked briefly on the tutorial, it needs some fixes:

1.
'metric_period':200,
'od_type':"Iter",
'od_wait':20, 

This is a weird combination. This means you only calculate metrics on every 200 iteration, you probably wanted to print it out overy 200 iteration instead. To do that use verbose=200.

Actually you can see that this is not something you shouled use, because there is a warining:
"Warning: Overfitting detector is active, thus evaluation metric is calculated
on every iteration. 'metric_period' is ignored for evaluation metric."

2.
'loss_function':'RMSE',
'eval_metric':'RMSE',

If loss_function is RMSE, eval_metric is RMSE by default, you don't need to set it.

3. 'learning_rate':0.001,
'depth':3

I'm not sure why this combination is used, I would suggest to first train with default parameters, then look if there is overfitting or underfitting and adjust learning_rate accordingly.
After that you can try changing other parameters and see if it improves the quality.

And for regression it might be useful to experiment with CTR settings, for example set TargetBorderCount to 2 or 3 instead of 1.

**Comment by dennislamcv1** [comment]
OK noted with thanks. That's what the instructor passed me this file.
